platform/core/src/types/HangingProtocol.ts

```diff
@@ -75,9 +75,9 @@ type ViewportStructure = {
  * it won't be selected.
  */
 type DisplaySetSelector = {
-  // The image matching rule (not currently implemented) selects which image to
-  // display initially, only for stack views.
-  imageMatchingRules?: MatchingRule[];
+  // The instance matching rules selects which instance to
```

Comment:
The description is correct, but the implementation is NOT agreeing with the description.  The implementation changes the internal representation of a display set, effectively removing elements from the display set.  The selectors are supposed to change what is displayed, not the underlying values, and that is what is causing all the problems with actually changing display sets.   Also, there are other types of bugs introduced by this description because it isn't at all clear that one cannot make selections from multiple series, for example:
Series Matching Rules   has Modality=SR as the match
Instance Matching Rules matches basic DICOM SR
expectation is to match ALL series containing DICOM SR instances.
There are also rules such as match one display set for each DICOM SR without a history object mapping it, which require custom code to implement.

Reply:
> The description is correct, but the implementation is NOT agreeing with the description. The implementation changes the internal representation of a display set, effectively removing elements from the display set. The selectors are supposed to change what is displayed, not the underlying values, and that is what is causing all the problems with actually changing display sets. 

This is incorrect. The implementation agrees with the description and the general logic of the hanging protocol matching rules.
You need to look at the `HangingProtocolService. _matchImages function to see the logic, but in summary,  if a "required" matching rule fails on a study, then that study is completely ignored/removed from the display, same thing for series, if a "required" matchingRule fails, the series is removed from the display, for this implementation to support "instanceMatchingRules" the implementation will always assume the rules "Required" and do the same on the instance level and REMOVE instances that fails the "InstanceMatchingRules" from the display

Otherwise, how would you describe the "InstanceMatchingRules" logic to work? 

> Also, there are other types of bugs introduced by this description because it isn't at all clear that one cannot make selections from multiple series, for example:
> Series Matching Rules has Modality=SR as the match
> Instance Matching Rules matches basic DICOM SR
> expectation is to match ALL series containing DICOM SR instances.
> There are also rules such as match one display set for each DICOM SR without a history object mapping it, which require custom code to implement.

I am not sure I completely follow your examples above, @wayfarer3130. How is the current implementation causes a bug in this case?
The SeriesMatchingRule will select the DisplaySets with the Modality=SR, what value is in the "InstanceMatchingRule" that will cause a filtering issue? 
I am not even sure why there would be a use-case where "InstanceMatchingRules"  are applicable with a SeriesMatchingRules of Modality `SR` but even if there is, then this is the user choice for defining these rules.